src/onnx_ir/_core_test.py

```diff
@@ -52,6 +52,7 @@ def test_init_requires_type_when_value_is_not_np_array(self):
             ("float8e4m3fnuz", np.uint8, ir.DataType.FLOAT8E4M3FNUZ),
             ("float8e5m2", np.uint8, ir.DataType.FLOAT8E5M2),
             ("float8e5m2fnuz", np.uint8, ir.DataType.FLOAT8E5M2FNUZ),
+            ("float8e8m0", ml_dtypes.float8_e8m0fnu, ir.DataType.FLOAT8E8M0),
```

Comment:
Here, we test when the input data is uint8 for FLOAT8E8M0. You may need to update the implementation in _core.py to handle when input data is uint8 and dtype is specified as FLOAT8E8M0 to behave correctly.

```suggestion
            ("float8e8m0", np.uint8, ir.DataType.FLOAT8E8M0),
```